### PR TITLE
Issue 35 list application versions

### PR DESCRIPTION
**Pull request by elauria**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.*.un~
```
Comment (prayerslayer) [inline .gitignore]: Out of curiosity, what does this?

File: client/lib/application/src/version-list/application-version.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import BaseView from 'common/src/base-view';
+import Template from './application-version.hbs';
+import Flux from 'application/src/flux';
+import FetchResult from 'common/src/fetch-result';
+import ErrorTpl from 'common/src/error.hbs';
+import Placeholder from '../detail/placeholder.hbs';
+import Markdown from 'common/src/markdown';
+
+import 'common/asset/scss/application/application.scss';
```
Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.js]: This you don’t need, right? Also the path doesn’t exist.

File: .gitignore (hunk 1 above)
Comment (elauria) [inline .gitignore]: to ignore vi tmp files :)

File: client/lib/application/src/version-list/application-version.js (hunk 2 above)
Comment (elauria) [inline client/lib/application/src/version-list/application-version.js]: right, missed this.

File: client/lib/application/src/router.js [hunk 3]
```diff
@@ -57,6 +59,20 @@ class AppRouter extends Router {
         .fetchApplications()
         .finally( () => puppeteer.show( new List(), MAIN_VIEW_ID ) );
     }
+
+    /**
+     * Fetches the application with `id`. Does not wait to finish and
```
Comment (prayerslayer) [inline client/lib/application/src/router.js]: It fetches versions, doesn’t it?

File: client/lib/application/src/version-list/application-version.hbs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<h2>{{app.name}} Versions</h2>
```
Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.hbs]: There is no `app` field in `state` tho.

File: client/lib/common/src/data/application/application-actions.js [hunk 5]
```diff
@@ -37,6 +37,18 @@ class ApplicationActions extends Actions {
                     throw err;
                 });
     }
+
+    fetchApplicationVersions(id) {
+        return request
+                .get(`${Services.kio.url}${Services.kio.root}/${id}/versions`)
```
Comment (prayerslayer) [inline client/lib/common/src/data/application/application-actions.js]: We should aim to do URL construction in `common/data/services`, so it’s easily* changeable. Will have a look later.

* Except for templates.

File: client/lib/application/src/router.js (hunk 3 above)
Comment (elauria) [inline client/lib/application/src/router.js]: true, now it does!

File: client/lib/common/src/data/application/application-store.js [hunk 6]
```diff
@@ -30,6 +31,13 @@ class ApplicationStore extends Store {
             this.beginSaveApplication,
             this.receiveApplication,
             this.failSaveApplication);
+
+        this.registerAsync(
```
Comment (prayerslayer) [inline client/lib/common/src/data/application/application-store.js]: Would be cool if we could also fetch and accept a single version.

File: client/lib/application/src/version-list/application-version.hbs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+{{#if versions.length}}
+    <ul>
+        {{#each versions}}
+            <li>
+                <a href="version/{{this.id}}">{{this.id}}</a>
```
Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.hbs]: This route doesn’t exist.

File: client/lib/application/src/version-list/application-version.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import BaseView from 'common/src/base-view';
+import Template from './application-version.hbs';
+import Flux from 'application/src/flux';
+import FetchResult from 'common/src/fetch-result';
```
Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.js]: Please remove unused imports.

File: client/lib/application/src/version-list/application-version.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import ErrorTpl from 'common/src/error.hbs';
+import Placeholder from '../detail/placeholder.hbs';
+import Markdown from 'common/src/markdown';
+
+import 'common/asset/scss/application/application.scss';
+
+class AppVersion extends BaseView {
+    constructor( props ) {
+        this.stores = {
+            application: Flux.getStore('application'),
```
Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.js]: Trailing comma.

File: client/lib/application/src/version-list/application-version.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        };
+        this.className = 'applicationVersion';
+        super(props);
+    }
+
+    update() {
+        let {applicationId} = this.props;
+        this.data = {
+            versions: this.stores.application.getApplicationVersions( applicationId ),
```
Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.js]: Trailing comma.

File: client/lib/application/src/version-list/application-version.hbs (hunk 7 above)
Comment (elauria) [inline client/lib/application/src/version-list/application-version.hbs]: yes, it will though, but you're right, shouldn't be pointing to it yet.

Comment (prayerslayer) [inline client/lib/application/src/version-list/application-version.hbs]: Thing is that we deploy quite often currently for different reasons. And when smth doesn’t work, people will complain about it.

Maybe we should add a big [beta stamp](http://www.mijndoelenstellen.nl/wp-content/uploads/2013/12/beta-stamp.png) as background-image.

**Comment by prayerslayer** [comment]
:bow:

**Pull request merged**